exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java

```diff
@@ -128,55 +148,184 @@ void createTestKitWithoutServices() {
+      TestSchema testSchema = testService.createDataSchema(view);
+      ProofMapIndexProxy<HashCode, String> proofMapIndexProxy = testSchema.testMap();
+      return toMap(proofMapIndexProxy);
+    });
+    assertThat(map).hasSize(1);
+    String initialValue = map.get(TestService.INITIAL_ENTRY_KEY);
+    assertThat(initialValue).isEqualTo(TestService.INITIAL_ENTRY_VALUE);
+  }
 
-    private static final TransactionConverter THROWING_TX_CONVERTER = (tx) -> {
-      throw new IllegalStateException("No transactions in this service: " + tx);
-    };
+  @Test
+  void createEmptyBlock() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
+    Block block = testKit.createBlock();
+    assertThat(block.getNumTransactions()).isEqualTo(0);
 
-    @Override
-    protected void configure() {
-      bind(Service.class).to(TestService.class).in(Singleton.class);
-      bind(TransactionConverter.class).toInstance(THROWING_TX_CONVERTER);
-    }
+    testKit.withSnapshot((view) -> {
+      Blockchain blockchain = Blockchain.newInstance(view);
+      assertThat(blockchain.getHeight()).isEqualTo(1);
+      assertThat(block).isEqualTo(blockchain.getBlock(1));
+      return null;
+    });
   }
 
-  public static final class TestServiceModule2 extends AbstractServiceModule {
+  @Test
+  void afterCommitSubmitsTransaction() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
 
-    private static final TransactionConverter THROWING_TX_CONVERTER = (tx) -> {
-      throw new IllegalStateException("No transactions in this service: " + tx);
-    };
+    Block block = testKit.createBlock();
+    List<TransactionMessage> inPoolTransactions = testKit
+        .findTransactionsInPool(tx -> tx.getServiceId() == TestService.SERVICE_ID);
+    assertThat(inPoolTransactions).hasSize(1);
+    TransactionMessage inPoolTransaction = inPoolTransactions.get(0);
+    RawTransaction afterCommitTransaction = constructAfterCommitTransaction(block.getHeight());
 
-    @Override
-    protected void configure() {
-      bind(Service.class).to(TestService2.class).in(Singleton.class);
-      bind(TransactionConverter.class).toInstance(THROWING_TX_CONVERTER);
-    }
+    assertThat(inPoolTransaction.getServiceId())
+        .isEqualTo(afterCommitTransaction.getServiceId());
+    assertThat(inPoolTransaction.getTransactionId())
+        .isEqualTo(afterCommitTransaction.getTransactionId());
+    assertThat(inPoolTransaction.getPayload()).isEqualTo(afterCommitTransaction.getPayload());
+
+    Block nextBlock = testKit.createBlock();
+    assertThat(nextBlock.getNumTransactions()).isEqualTo(1);
+    assertThat(nextBlock.getHeight()).isEqualTo(2);
   }
 
-  static final class TestService implements Service {
+  @Test
+  void createBlockWithTransactionIgnoresInPoolTransactions() {
+    TestKit testKit = TestKit.forService(TestServiceModule.class);
 
-    static short SERVICE_ID = 46;
-    static String SERVICE_NAME = "Test service";
+    testKit.createBlock();
```

Comment:
It is unclear why this is invoked.

Reply:
We want to create a block so that `afterCommit` submits a transaction into the pool and when we create the block with specified transactions, it ignores this in-pool transaction - then we check that it is indeed still in the pool (together with the second `afterCommit` transaction)